### PR TITLE
Show warning when leaving multiple editors with unsaved changes

### PR DESCRIPTION
**Pull request by ClementPasteau**
Improving the UX of Dialogs where one can change a lot of properties and lose them if misclicking on the Cancel button.

https://user-images.githubusercontent.com/4895034/199949015-0e7e5d44-44af-4850-95f0-4022c9c3f344.mov

### CODE REVIEW

File: newIDE/app/src/BehaviorsEditor/index.js [hunk 1]
```diff
@@ -155,6 +159,8 @@ const BehaviorsEditor = (props: Props) => {
                           onClick={ev => {
                             ev.stopPropagation();
                             onRemoveBehavior(behaviorName);
+                            if (props.onBehaviorsUpdated)
+                              props.onBehaviorsUpdated();
```
Comment (AlexandreSi) [inline newIDE/app/src/BehaviorsEditor/index.js:163]: Isn't it already handled by onRemoveBehavior method?

File: newIDE/app/src/BehaviorsEditor/index.js [hunk 2]
```diff
@@ -213,6 +219,8 @@ const BehaviorsEditor = (props: Props) => {
                         onClick={ev => {
                           ev.stopPropagation();
                           onRemoveBehavior(behaviorName);
+                          if (props.onBehaviorsUpdated)
+                            props.onBehaviorsUpdated();
```
Comment (AlexandreSi) [inline newIDE/app/src/BehaviorsEditor/index.js:223]: Same here

File: newIDE/app/src/VariablesList/VariablesList.js [hunk 3]
```diff
@@ -291,6 +292,7 @@ const VariablesList = ({ onComputeAllVariableNames, ...props }: Props) => {
     props.historyHandler
       ? props.historyHandler.saveToHistory()
       : setHistory(saveToHistory(history, props.variablesContainer));
+    if (props.onVariablesUpdated) props.onVariablesUpdated();
```
Comment (AlexandreSi) [inline newIDE/app/src/VariablesList/VariablesList.js:295]: Should the method be renamed "onChange" ?

File: newIDE/app/src/ObjectGroupEditor/index.js [hunk 4]
```diff
@@ -114,9 +124,9 @@ const ObjectGroupEditor = ({
             project={project}
             globalObjectsContainer={globalObjectsContainer}
             objectsContainer={objectsContainer}
-            value={newObjectName}
+            value={objectName}
             excludedObjectOrGroupNames={objectsInGroup}
-            onChange={setNewObjectName}
+            onChange={editObject}
```
Comment (AlexandreSi) [inline newIDE/app/src/ObjectGroupEditor/index.js:129]: I think this is only to set the textfield value. So you could only use `setObjectName` and not call the `onObjectGroupUpdted`

File: newIDE/app/src/ProjectManager/LoadingScreenEditor.js [hunk 5]
```diff
@@ -23,19 +23,21 @@ import SelectField from '../UI/SelectField';
 import SelectOption from '../UI/SelectOption';
 import Text from '../UI/Text';
 
-type Props = {
+type Props = {|
```
Comment (AlexandreSi) [inline newIDE/app/src/ProjectManager/LoadingScreenEditor.js:26]: 👍

File: newIDE/app/src/ProjectManager/LoadingScreenEditor.js [hunk 6]
```diff
@@ -112,15 +134,19 @@ export const LoadingScreenEditor = ({
               }
               loadingScreen.showGDevelopSplash(checked);
               forceUpdate();
+              onLoadingScreenUpdated();
```
Comment (AlexandreSi) [inline newIDE/app/src/ProjectManager/LoadingScreenEditor.js:137]: Prefer using `onUpdate`

File: newIDE/app/src/ProjectManager/LoadingScreenEditor.js [hunk 7]
```diff
@@ -112,15 +134,19 @@ export const LoadingScreenEditor = ({
...
             floatingLabelText={<Trans>GDevelop logo style</Trans>}
             value={loadingScreen.getGDevelopLogoStyle()}
-            onChange={(e, i, value: string) => {
-              loadingScreen.setGDevelopLogoStyle(value);
+            onChange={(e, i, newGdevelopLogoStyle: string) => {
+              const currentGDevelopLogoStyle = loadingScreen.getGDevelopLogoStyle();
+              if (currentGDevelopLogoStyle === newGdevelopLogoStyle) return;
+              loadingScreen.setGDevelopLogoStyle(newGdevelopLogoStyle);
               forceUpdate();
+              onLoadingScreenUpdated();
```
Comment (AlexandreSi) [inline newIDE/app/src/ProjectManager/LoadingScreenEditor.js:149]: use `onUpdate`here too

File: newIDE/app/src/ProjectManager/LoadingScreenEditor.js [hunk 8]
```diff
@@ -223,21 +293,31 @@ export const LoadingScreenEditor = ({
...
-              );
+            onChange={newValue => {
+              const currentProgressBarHeight = loadingScreen.getProgressBarHeight();
+              const newProgressBarHeight = Math.max(1, parseFloat(newValue));
+              if (currentProgressBarHeight === newProgressBarHeight) {
+                return;
+              }
+              loadingScreen.setProgressBarHeight(newProgressBarHeight);
               forceUpdate();
+              onLoadingScreenUpdated();
```
Comment (AlexandreSi) [inline newIDE/app/src/ProjectManager/LoadingScreenEditor.js:304]: use `onUpdate` here too

File: newIDE/app/src/ProjectManager/LoadingScreenEditor.js [hunk 9]
```diff
@@ -223,21 +293,31 @@ export const LoadingScreenEditor = ({
...
-              loadingScreen.setProgressBarColor(rgbStringToHexNumber(color));
+            onChange={newColor => {
+              const currentProgressBarColor = loadingScreen.getProgressBarColor();
+              const newProgressBarColor = rgbStringToHexNumber(newColor);
+              if (currentProgressBarColor === newProgressBarColor) {
+                return;
+              }
+              loadingScreen.setProgressBarColor(newProgressBarColor);
               forceUpdate();
+              onLoadingScreenUpdated();
```
Comment (AlexandreSi) [inline newIDE/app/src/ProjectManager/LoadingScreenEditor.js:320]: ditto

File: newIDE/app/src/ProjectManager/LoadingScreenEditor.js [hunk 10]
```diff
@@ -251,9 +331,15 @@ export const LoadingScreenEditor = ({
...
-              loadingScreen.setMinDuration(Math.max(0, parseFloat(value)));
+            onChange={newValue => {
+              const currentMinDuration = loadingScreen.getMinDuration();
+              const newMinDuration = Math.max(0, parseFloat(newValue));
+              if (currentMinDuration === newMinDuration) {
+                return;
+              }
+              loadingScreen.setMinDuration(newMinDuration);
               forceUpdate();
+              onLoadingScreenUpdated();
```
Comment (AlexandreSi) [inline newIDE/app/src/ProjectManager/LoadingScreenEditor.js:342]: ditto

**Review by AlexandreSi — APPROVED** [review]
LGTM after comments are addressed

File: newIDE/app/src/BehaviorsEditor/index.js (hunk 1 above)
Comment (ClementPasteau) [reply newIDE/app/src/BehaviorsEditor/index.js:163]: yes good catch!

File: newIDE/app/src/VariablesList/VariablesList.js (hunk 3 above)
Comment (ClementPasteau) [reply newIDE/app/src/VariablesList/VariablesList.js:295]: I wasn't sure, it felt to me like `onChange` is really generic

File: newIDE/app/src/ObjectGroupEditor/index.js (hunk 4 above)
Comment (ClementPasteau) [reply newIDE/app/src/ObjectGroupEditor/index.js:129]: But technically it's a field like others no? Why wouldn't we want to warn if they changed it?

Comment (ClementPasteau) [reply newIDE/app/src/ObjectGroupEditor/index.js:129]: oh no you're right, it doesn't make sense

File: newIDE/app/src/VariablesList/VariablesList.js (hunk 3 above)
Comment (AlexandreSi) [reply newIDE/app/src/VariablesList/VariablesList.js:295]: Sorry, I was talking about the method `_saveToHistory`

**Pull request merged**